### ISSUE TITLE
`omit_defaults` does not omit tuples and frozensets

### ISSUE THREAD

**Issue by jodal** [jcrist/msgspec#652]
### Description

When a struct has `omit_defaults=True` set, any fields defaulting to `dict`, `list`, or `set` is omitted from the serialization, as described [in the docs](https://jcristharif.com/msgspec/structs.html#omitting-default-values).

However, if one is using `tuple` instead of `list` and `frozenset` instead of `set` as default values, to make the struct as immutable as possible, those defaults are included in the encoded format.

I think that `tuple` and `frozenset` should be handled in the same way as `dict`, `list, and `set` by `omit_defaults` and `repr_omit_defaults`.

### Example

```python
import msgspec


class Backpack(msgspec.Struct, omit_defaults=True):
    a_list: list[str] = msgspec.field(default_factory=list)
    a_tuple: tuple[str, ...] = msgspec.field(default_factory=tuple) 
    a_frozenset: frozenset[str] = msgspec.field(default_factory=frozenset) 


instance = Backpack()
print(msgspec.json.encode(instance))
```

Output:

```
b'{"a_tuple":[],"a_frozenset":[]}'
```

Expected:

```
b'{}'
```

**Comment by jcrist** [comment]
Thanks for opening this - this should be fixed by #653. Note that for these cases specifying the default value by value (rather than via `default_factory`) already works:

```python
import msgspec


class Backpack(msgspec.Struct, omit_defaults=True):
    a_list: list[str] = []
    a_tuple: tuple[str, ...] = ()
    a_frozenset: frozenset[str] = frozenset()


instance = Backpack()
print(msgspec.json.encode(instance))
#> b'{}'
```

The reason `tuple` and `frozenset` are the special cases here is that they're both immutable types and may always be specified by value. Since there's no functional reason to use a `default_factory` for these types, we hadn't implemented the logic to check for equivalent defaults based on `default_factory` for `tuple`/`frozenset`.

**Comment by jodal** [comment]
Thank you for the very swift fix! Especially useful with the tip on how to achieve this without waiting for a new release :-)
